File: go/strategy/strategy_test.go
package strategy

import (
	"design_pattern/strategy/fee"
	"design_pattern/strategy/remit"
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestStrategy(t *testing.T) {
	t.Run("Strategy run", func(t *testing.T) {
		var amount float64 = 1000
		defaultFee := amount * fee.DefaultFee
		partnershipFee := amount * fee.PartnershipFee

		abcBankRemit := remit.NewAbcBankRemit(fee.NewDefaultFee())
		defBankRemit := remit.NewDefBankRemit(fee.NewPartnershipFee())

		amountAbc := abcBankRemit.CalculateAmount(amount)
		amountDef := defBankRemit.CalculateAmount(amount)

		assert.Equal(t, amount+defaultFee, amountAbc)
		assert.Equal(t, amount+partnershipFee, amountDef)
	})
}

File: typescript/strategy/strategy.test.ts
import { AbcBankRemit } from "./abc-bank-remit";
import { DefBankRemit } from "./def-bank-remit";
import { DefaultFee } from "./fee/default-fee";
import { PartnershipFee } from "./fee/partnership-fee";

test('Strategy Run', () => {
    const amount = 10000;
    const defaultFee = amount * DefaultFee.FEE;
    const partnershipFee = amount * PartnershipFee.FEE;
    
    const abcBankRemit = new AbcBankRemit(new DefaultFee());
    const defBankRemit = new DefBankRemit(new PartnershipFee());

    const amountAbc = abcBankRemit.caculateAmount(amount);
    const amountDef = defBankRemit.caculateAmount(amount);

    expect(amountAbc).toEqual(amount + defaultFee);
    expect(amountDef).toEqual(amount + partnershipFee);
});